src/k8s-engine/wsl.ts

```diff
@@ -1306,6 +1309,50 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     await this.execCommand('/usr/sbin/update-ca-certificates');
   }
 
+  /**
+   * Run provisioning scripts; this is done after init is started.
+   */
+  protected async runProvisioningScripts() {
+    const provisioningPath = path.join(paths.config, 'provisioning');
+
+    await fs.promises.mkdir(provisioningPath, { recursive: true });
+    await Promise.all([
+      (async() => {
+        // Write out the readme file.
+        const ReadmePath = path.join(provisioningPath, 'README');
+
+        try {
+          await fs.promises.access(ReadmePath, fs.constants.F_OK);
+        } catch {
+          const contents = `
+            Any files named '*.start' in this directory will be executed
+            sequentially on Rancher Desktop startup, before the main services.
+            Files are processed in lexical order, and will delay startup until
+            they are complete. Similaryly, any files named '*.stop' will be
+            executed on shutdown, after the main services have exited.
+            `.replace(/\s*\n\s*/g, '\n').trim();
+
+          await fs.promises.writeFile(ReadmePath, contents, { encoding: 'utf-8' });
+        }
+      })(),
+      (async() => {
+        const linuxPath = await this.wslify(provisioningPath);
+
+        // Run the provisioning job. We need to clobber /etc/init.d as
+        // /etc/init.d/local has no options to use a different directory.
+        await this.execCommand('/bin/rm', '-r', '-f', '/etc/local.d');
+        await this.execCommand('/bin/ln', '-s', '-f', '-T', linuxPath, '/etc/local.d');
+        await this.execCommand(
+          '/bin/sh',
+          '-c',
+          `for f in '${ linuxPath }'/*.start; do [ -e "\${f}" ] && chmod a+x "\${f}"; done`);
```

Comment:
Not sure why the `-e $f` test is needed; how would it show up in the glob if it doesn't exist?

And if the test is indeed needed, shouldn't it be `-f` instead of `-e`?

Reply:
Also, why is this only done for `*.start` scripts, but not for `*.stop`?

Reply:
If there are no matches, we get a raw `*.start` back (because busybox ash doesn't have a `nullglob` option.

We do need it for `*.stop` too (it seems to magically work, but I don't trust it so I'll fix).